BinaryEncoding.md

```diff
@@ -584,6 +584,8 @@ natural alignment. The bits after the
 are reserved for [future :unicorn:][future threads] use
 (e.g., for shared memory ordering requirements).
 
+It is a validation error if `offset + numBytesTouched(load/store) - 1 > 2^32 - 1`. If this condition is true, the load or store is guaranteed to be out of bounds.
```

Comment:
Emphasizing how this catches accesses "guaranteed to be out of bounds" could seem to suggest this feature is intended to help catch bugs in application code. However, for reasons you noted earlier, the specific check here wouldn't be very effective for that purpose. To me, the more important motivation for this check is eliminating certain special cases that would arise as a result of integer overflows.

So perhaps change "If this condition is true, the load or store is guaranteed to be out of bounds" to "This frees consumers from some special cases that could arise from integer overflows in addressing arithmetic" or so.